libclc/generic/libspirv/atomic/atomic_cmpxchg.cl

```diff
@@ -24,6 +24,13 @@ _Z29__spirv_AtomicCompareExchangePU3AS1iN5__spv5Scope4FlagENS1_19MemorySemantics
   return __sync_val_compare_and_swap(p, cmp, val);
 }
 
+_CLC_DEF int
+_Z29__spirv_AtomicCompareExchangePiN5__spv5Scope4FlagENS0_19MemorySemanticsMask4FlagES4_ii(
```

Comment:
Do these definitions also need to be wrapped in `_CLC_GENERIC_AS_SUPPORTED`?

Reply:
Yeah well spotted, thank you. I'd also to refactor this file to use a macro system similar to the other files, but I'd rather do that in a separate PR

Reply:
Yeah, there's a lot of opportunity for that sort of thing around libclc - enough work for a good while :)